pl_bolts/datasets/base_dataset.py

```diff
@@ -58,3 +63,31 @@ def _download_from_url(self, base_url: str, data_folder: str, file_name: str):
             urllib.request.urlretrieve(url, fpath)
         except HTTPError as err:
             raise RuntimeError(f"Failed download from {url}") from err
+
+
+@dataclass
+class DataModel:
+    """Data model dataclass.
+
+    Ties together data and callable transforms.
+
+    Attributes:
+        data: Sequence of indexables.
+        transform: Callable to transform data.
+    """
+
+    data: ARRAYS
+    transform: Optional[Callable] = None
+
+    def process(self, data: ARRAYS) -> ARRAYS:
```

Comment:
Why does it need an argument `data`? And if that's there and it's to serve only as transform, then having `data` as an instance attribute doesn't make much sense. Or am I missing something?